SME1 support for int4 GEMM and GEMMV kernels

qti-vaiskv opened the pull request:
Adds support for SME1 int4 GEMM and GEMMV kernels.

qti-vaiskv committed:
Added Qualcomm Kleidiai library support
 Qualcomm Kleidiai library supports  QMX specific kernels

qti-vaiskv committed:
Added support for SME1 qp8_f32_qc4w GEMM kernel

dsharlet reviewed (commented):
It looks like this is adding a new dependency, a fork of KleidiAI? https://github.com/qualcomm/kleidiai

We can't have two versions of the same library in our repo. I'm surprised it works at all.

Is it even necessary? https://github.com/qualcomm/kleidiai doesn't appear to have any commits that aren't in the upstream repo.

qti-vaiskv commented:
There are few missing SME1 kernels especially the int4 datatypes in the parent Kleidiai. We have added support for these kernels in Qualcomm's fork of Kleidiai.

We are maintaining this repo to account for these missing kernels and there will not be any duplicate definitions of the functions that are already part of Kleidiai. 

https://github.com/qualcomm/kleidiai/commits/kleidiai-qmx-xnnpack/

**Note:** In this PR we have added only the cmake support. We will add the bazel support as well in this same PR.

qti-vaiskv commented:
Kleidiai has added an optimized SME1 kernel for qp8-f32-qc4w GEMM case.  Closing is draft PR since the changes made here to pull new dependency (Qualcomm fork of Kleidiai) are not required. 

 Created a new PR #10230
